interfaces/policy/policy.go

```diff
@@ -168,37 +168,47 @@ func (connc *ConnectCandidate) slotPublisherID() string {
 	return "" // never a valid publisher-id
 }
 
-func (connc *ConnectCandidate) checkPlugRule(rule *asserts.PlugRule, snapRule bool) error {
+func (connc *ConnectCandidate) checkPlugRule(kind string, rule *asserts.PlugRule, snapRule bool) error {
 	context := ""
 	if snapRule {
 		context = fmt.Sprintf(" for %q snap", connc.PlugSnapDeclaration.SnapName())
 	}
-	if checkPlugConnectionConstraints(connc, rule.DenyConnection) == nil {
-		return fmt.Errorf("connection denied by plug rule of interface %q%s", connc.Plug.Interface, context)
+	denyCstrsSel := asserts.ConstraintsDenyConnection
+	allowCstrsSel := asserts.ConstraintsAllowConnection
+	if kind == "auto-connection" {
+		denyCstrsSel = asserts.ConstraintsDenyAutoConnection
+		allowCstrsSel = asserts.ConstraintsAllowAutoConnection
```

Comment:
While reading this I'm wondering a bit whether we need those enums, instead of using the fields directly. E.g.
```
denyCons = &rule.DenyAutoConnection
```
Feels like the indirection through Get* is unnecessary.

(side note: cstrs is hard to type and to read; I suggest "const" as an alternative abbreviation)